pkg/planner/core/plan_clone_generated.go

```diff
@@ -180,3 +180,104 @@ func (op *PhysicalMergeJoin) CloneForPlanCache(newCtx base.PlanContext) (base.Pl
 	cloned.basePhysicalJoin = *basePlan
 	return cloned, true
 }
+
+// CloneForPlanCache implements the base.Plan interface.
+func (op *PhysicalTableReader) CloneForPlanCache(newCtx base.PlanContext) (base.Plan, bool) {
+	cloned := new(PhysicalTableReader)
+	*cloned = *op
+	basePlan, baseOK := op.physicalSchemaProducer.cloneForPlanCacheWithSelf(newCtx, cloned)
+	if !baseOK {
+		return nil, false
+	}
+	cloned.physicalSchemaProducer = *basePlan
+	tablePlan, ok := op.tablePlan.CloneForPlanCache(newCtx)
+	if !ok {
+		return nil, false
+	}
+	cloned.tablePlan = tablePlan.(base.PhysicalPlan)
+	cloned.TablePlans = flattenPushDownPlan(cloned.tablePlan)
+	cloned.PlanPartInfo = op.PlanPartInfo.Clone()
+	if op.TableScanAndPartitionInfos != nil {
```

Comment:
if TableScanAndPartitionInfos != nil, refuse clone?

Reply:
YES, this field is only for MPP, but MPP plan is not supported by Plan Cache.